src/workload.py

```diff
@@ -27,30 +27,31 @@ class KafkaWorkload(WorkloadBase):
 
     # FIXME: Paths and constants integrated into WorkloadBase?
     SNAP_NAME = "charmed-kafka"
-    SNAP_SERVICE = "daemon"
     LOG_SLOT = "logs"
 
-    def __init__(self) -> None:
+    def __init__(self, role: Role) -> None:
```

Comment:
**suggestion, major** Also here I would honestly use two different classes to deal with Kafka and CruiseControl, say `KafkaWorkload` and `CruiseControlWorkload`. To me, they are two different workloads, and they own two different classes.

Reply:
I kind of disagree here because they are so close that we will just duplicate ~90% of the class, or add another abstraction layer on top of `WorkloadBase`.

Reply:
What do you think about injecting the service itself instead of passing `role`?

Reply:
If the class were very general, and be parametric with respect to only the name of the service, I would change the parameter name to service. 

However, In this specific case, I would opt for inheritance. Also because I can see some differences with scope (they are two different processes) but also functionalities, like [here](https://github.com/canonical/kafka-operator/blob/85e230fb88e2a67723da60602c7895ce787ad03a/src/workload.py#L171), which apply only to one workload. If you have two classes you could handle more cleanly these custom methods, their different implementation (without branching logic) but also whether they exists or not (that helps with linting/typing). 

Again, you should see this more specific point related to the general pattern (suggestion) of not passing down the `role` attribute.